src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/WindowsRuntime/Generated/WinRT/GuidGenerator.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace WinRT
+{
+    internal static class GuidGenerator
+    {
+        public static Guid GetGUID(Type type)
+        {
+            return type.GetGuidType().GUID;
+        }
+
+        public static Guid GetIID(Type type)
+        {
+            type = type.GetGuidType();
+            if (!type.IsGenericType)
+            {
+                return type.GUID;
+            }
+            return (Guid)type.GetField("PIID").GetValue(null);
+        }
+
+        public static string GetSignature(Type type)
+        {
+            var helperType = type.FindHelperType();
+            if (helperType != null)
+            {
+                var sigMethod = helperType.GetMethod("GetGuidSignature", BindingFlags.Static | BindingFlags.Public);
+                if (sigMethod != null)
+                {
+                    return (string)sigMethod.Invoke(null, new Type[] { });
+                }
+            }
+
+            if (type == typeof(object))
+            {
+                return "cinterface(IInspectable)";
+            }
+
+            if (type.IsGenericType)
+            {
+                var args = type.GetGenericArguments().Select(t => GetSignature(t));
+                return "pinterface({" + GetGUID(type) + "};" + String.Join(";", args) + ")";
```

Comment:
I recommend using `string` instead of `String`

In [runtime/coding-style.md at master · dotnet/runtime](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md ) NO.11, we use language keywords instead of BCL types